src/concepts/rendering-under-the-hood.md

```diff
@@ -0,0 +1,265 @@
+    buffer: &'a mut Buffer,
+}
+```
+
+The `Buffer` represents the area in which content is going to be drawn to the terminal. The `Buffer`
+struct implements a number of methods such as [`get_mut`] and [`set_string`]:
+
+```rust
+pub fn get_mut(&mut self, x: u16, y: u16) -> &mut Cell {
+}
+
+pub fn set_string<S>(&mut self, x: u16, y: u16, string: S, style: Style)
+where
+    S: AsRef<str> {
+}
+```
+
+These methods allow any implementation of the `Widget` trait to write into any parts of the
+`Buffer`.
+
+A [`Buffer`] is essentially a collection of `Cell`s:
+
+```rust
+pub struct Buffer {
+    // --snip--
+    pub content: Vec<Cell>,
+}
+```
+
+In Ratatui, this `Cell` struct is the smallest renderable unit of code. A `Cell` essentially has a
+`String` which is the symbol that will be drawn in that `Cell` as well as `Style` information
+(foreground color, background color, modifiers etc).
+
+```rust
+// pseudo code
+pub struct Cell {
+  content: String,
+  style: Style,
+}
+```
+
+```admonish
+`Cell`s are the closest analog to a "pixel" in a terminal.
+Most terminals render monospaced text 2 high and 1 wide.
+A `Cell` in Ratatui should usually contain 1 wide string content.
+```
+
+Here's a simplified class diagram for reference:
+
+```mermaid
+classDiagram
+    class Cell {
+        +content: String
+        +style: Style
+    }
+
+    class Buffer {
+        +area: Rect
+        +content: Vec<Cell>
+    }
+
+    class Frame {
+        +buffer: mut Buffer
+        +render_widget()
+    }
+
+    class Widget {
+        <<interface>>
+        +render(area: Rect, buf: &mut Buffer)
+    }
+
+    class Paragraph
+    class Block
+    class Table
+
+    Buffer --> Cell
+    Frame --> Buffer
+    Widget <|.. Paragraph
+    Widget <|.. Block
+    Widget <|.. Table
+
+
+```
+
+Basically what this all means that any content rendered to a `Buffer` is only stored in that
+particular `Buffer` struct in the form of `Cell`s that contain string and style information.
+
+```admonish note
+Because any styling has to happen though the `Style` struct (i.e. methods that apply
+`Style`s to `Cell`s) you can't use ansi escape sequences directly in the string content.
+
+As a workaround, what you can do however is convert the ansi escape sequences into a
+`ratatui::text::Text` struct that contains the appropriate `Style`s. There's a crate for this that
+might help: <https://crates.io/crates/ansi-to-tui>. And this `Text` struct that you create can be
+passed into the `Paragraph` widget to get the desired effect.
+
+Even if this particular crate doesn't work for you this approach is probably what you want to use if
+you want to render ansi escape sequences in Ratatui.
+You can learn more about the text related of Ratatui features and displaying text
+[here](./../how-to/render/display-text.md).
+```
+
+After all the `Widget`s are rendered to the same `Buffer` struct, i.e. after
+`terminal.draw(|f| ...)` returns, [`terminal.flush()`] is called. This is where the content is
```

Comment:
After the closure provided to the `draw` method finishes, the `draw` method calls `Terminal::flush()`. This causes the buffer to be written to the screen. Ratatui uses a double buffer approach so that <REASON:  usually this is something like "the process of writing from the buffer to the screen happens all at once and the next buffer can be constructed while the first one is being rendered, but this isn't something that actually happens AFAIK>